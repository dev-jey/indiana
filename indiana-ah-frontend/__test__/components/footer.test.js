import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Footer from '../../src/components/common/footer.jsx';

describe('Custom Footer component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <Footer />
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders a footer', () => {
    const footer = shallow(<Footer />);
    expect(footer.find('.footer')).toBeDefined();
    expect(footer.find('.app-logo')).toBeDefined();
    const h2 = footer.find('h2');
    expect(h2.length).toEqual(3);
  });
});
