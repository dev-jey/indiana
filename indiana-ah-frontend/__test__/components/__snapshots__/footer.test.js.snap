// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Footer component renders correctly 1`] = `
<div
  className="footer"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-4 col-md-5 col-sm-4 col-3 d-none d-sm-block pl-0 align-self-center col"
      >
        <div
          className="d-flex"
        >
          <img
            alt="facebook logo"
            className="feather"
            src="test-file-stub"
          />
          <img
            alt="facebook logo"
            className="d-block app-logo"
            src="test-file-stub"
          />
        </div>
      </div>
      <div
        className="d-flex align-items-center flex-column col"
      >
        <h2
          className="mb-5"
        >
          Help
        </h2>
        <a
          href="/"
          onClick={[Function]}
        >
          <h4
            className="mb-4"
          >
            Profile
          </h4>
        </a>
        <a
          href="/"
          onClick={[Function]}
        >
          <h4>
            Blog
          </h4>
        </a>
      </div>
      <div
        className="d-flex align-items-center flex-column col"
      >
        <h2
          className="mb-5"
        >
          Contact Us
        </h2>
        <a
          href="/"
          onClick={[Function]}
        >
          <h4
            className="mb-4"
          >
            Faqs
          </h4>
        </a>
        <a
          href="/"
          onClick={[Function]}
        >
          <h4>
            Contact us
          </h4>
        </a>
      </div>
      <div
        className="d-flex align-items-center flex-column col"
      >
        <h2
          className="mb-4"
        >
          Social Media
        </h2>
        <div
          className="d-flex justify-evenly w-75 mb-4"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="facebook logo"
              className="mr-6 d-block logo2"
              src="test-file-stub"
            />
          </a>
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="twitter logo"
              className="d-block logo3"
              src="test-file-stub"
            />
          </a>
        </div>
        <div
          className="d-flex justify-content-between w-75"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <h4>
              Facebook
            </h4>
          </a>
          <a
            href="/"
            onClick={[Function]}
          >
            <h4>
              Twitter
            </h4>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
